src/Main.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from "react";
+import { Outlet } from "react-router-dom";
+import styled from "styled-components";
+import { getUser } from "./api";
+import Footer from "./components/common/Footer";
+import Header from "./components/common/Header";
+import useAsync from "./components/hooks/useAsync";
+
+const Container = styled.div`
+  min-height: 100%;
+`;
+
+interface UserInfo {
+  id: number;
+  name: string;
+  email: string;
+  image_source: string;
+}
+
+function Main() {
+  const [userInfo, setUserInfo] = useState<UserInfo>({
+    id: 0,
+    name: "",
+    email: "",
+    image_source: "",
+  });
+  const [, getUserAsync] = useAsync(getUser);
+
+  // 유저 정보 요청
+  const handleLoadUser = async () => {
+    if (typeof getUserAsync === "function") {
+      let result = await getUserAsync();
+      if (!result) return;
+      setUserInfo(result.data[0]);
+    }
```

Comment:
data가 배열인지, 또한 원하는 요소가 실제로 존재하는지 아래처럼 확인하는게 좋겠습니다.

```suggestion
if (result?.data?.length > 0) {
  setUserInfo(result.data[0]);
}
```